sql/core/src/test/scala/org/apache/spark/sql/QueryTest.scala

```diff
@@ -205,6 +205,20 @@ abstract class QueryTest extends PlanTest {
         planWithCaching)
   }
 
+  /**
+    * Asserts that a given [[Dataset]] will be executed using the given named cache.
+    */
+  def assertCached(query: Dataset[_], cachedName: String): Unit = {
+    val planWithCaching = query.queryExecution.withCachedData
+    val cachedTableNames = planWithCaching collect {
+      case cached: InMemoryRelation  => cached.cacheBuilder.tableName
```

Comment:
The result type of the `tableName` is `Option[String]`:

https://github.com/apache/spark/blob/ca13fc4649a8ef59e12c73a50f4cda7f1980077f/sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/InMemoryRelation.scala#L46-L51

So the `collect` will produce `Seq[Option[String]]` and will contain tablenames as `Some[String]` so this assert will be raised always as `cachedName` is `String`:
https://github.com/apache/spark/blob/ca13fc4649a8ef59e12c73a50f4cda7f1980077f/sql/core/src/test/scala/org/apache/spark/sql/QueryTest.scala#L218

Is not it?

Reply:
Thanks. I updated the method 'assertCached' to flatten those table names. We should not have type issue anymore.